js/errors.js

```diff
@@ -32,15 +32,19 @@
      * @param  {string} redirectPath path that would be added to the host to create full redirect link in Chaise
      * @constructor
      */
-    function ERMrestError(code, status, message, subMessage, redirectPath) {
+    function ERMrestError(code, status, message, subMessage, redirectPath, reference) {
+        this.errorData = {};
         this.code = code;
         this.status = status;
         this.message = message;
         this.subMessage = subMessage;
-        if(redirectPath !== undefined && redirectPath !== null){
-           this.errorData = {};
+        if(redirectPath !== undefined && redirectPath !== null) {
            this.errorData.redirectPath = redirectPath;
         }
+
+        if(reference !== undefined && reference !== null) {
+           this.errorData.reference = reference;
```

Comment:
I think just adding the reference to the `DuplicateConflictError` for now is enough. And name it something more specific. Like the `duplicateReference` name that you've chosen. Just `ERMrestError.reference` is confusing as what is this reference of.

I understand that you've added it for future usage. But I think this will just be more confusing and we should be more specific about these optional parameters.